v2/api/keyvault/customizations/vault_extensions.go

```diff
@@ -184,25 +186,36 @@ func (ex *VaultExtension) handlePurgeThenCreate(
 	return nil
 }
 
+type deletionDetails struct {
```

Comment:
I like the use of a result struct.

Reply:
Yeah I wrote it without one (partially) initially and the nil-checking was pretty awkward so went with this instead.